### ISSUE TITLE
[WDRPage] newer episodes from https://www.wdrmaus.de/ don’t work anymore

### ISSUE THREAD

**Issue by w4grfw** [yt-dlp/yt-dlp#7983]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.07.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

DE

### Provide a description that is worded well enough to be understood

Newer  episodes i.e. [https://www.wdrmaus.de/filme/sachgeschichten/haferdrink.php5](url) or [https://www.wdrmaus.de/filme/sachgeschichten/tartanbahn.php5](url) don’t work anymore. A working episode is [https://www.wdrmaus.de/filme/sachgeschichten/knatterboot.php5](url).
Maybe it’s caused by changing `class="videoButton play srtvd" rel="nofollow" data-extension="{ 'mediaObj': …` (this is from [https://www.wdrmaus.de/filme/sachgeschichten/tartanbahn.php5](url) to `class="videoButton play srtvd" rel="nofollow" data-extension-ard="{ 'mediaObj': …` (this ist from [https://www.wdrmaus.de/filme/sachgeschichten/haferdrink.php5](url).

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
$ ./yt-dlp --verbose --update --no-config https://www.wdrmaus.de/filme/sachgeschichten/haferdrink.php5
[debug] Command-line config: ['--verbose', '--update', '--no-config', 'https://www.wdrmaus.de/filme/sachgeschichten/haferdrink.php5']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481] (zip)
[debug] Lazy loading extractors is disabled
[debug] Python 3.11.4 (CPython x86_64 64bit) - Linux-6.2.0-31-generic-x86_64-with-glibc2.37 (OpenSSL 3.0.8 7 Feb 2023, glibc 2.37)
[debug] exe versions: ffmpeg 5.1.2 (setts), ffprobe 5.1.2, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.11.0, brotli-1.0.9, certifi-2022.09.24, mutagen-1.46.0, pyxattr-0.8.0, secretstorage-3.3.3, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1864 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
Current Build Hash: b8f4393fadb79fa3bc72f8415dc8a807c0222684c4a2ef57aade01f8eb0e65d3
yt-dlp is up to date (stable@2023.07.06)
[debug] Using fake IP 53.196.139.75 (DE) as X-Forwarded-For
[WDRPage] Extracting URL: https://www.wdrmaus.de/filme/sachgeschichten/haferdrink.php5
[WDRPage] haferdrink: Downloading webpage
[download] Downloading playlist: haferdrink
[WDRPage] Playlist haferdrink: Downloading 0 items
[download] Finished downloading playlist: haferdrink
```

**Comment by rdamas** [comment]
Yes, that's exactly the issue; the regex for extracting the ``mediaObj`` needs to be extended. Currently it only looks for ``data-extension=``, and it needs to recognize ``data-extension-ard`` as well.

**Comment by bashonly** [comment]
Same root issue as #7461

Will be fixed by #7979
